pkg/client/schema.go

```diff
@@ -3,34 +3,36 @@ package client
 import (
 	"context"
 
+	"github.com/cloudquery/cloudquery/pkg/plugin/registry"
+
 	"github.com/rs/zerolog/log"
 
 	"github.com/cloudquery/cloudquery/pkg/plugin"
 	"github.com/cloudquery/cq-provider-sdk/cqproto"
 )
 
 type GetProviderSchemaOptions struct {
-	Provider string
+	Provider registry.Provider
 }
 
 type ProviderSchema struct {
 	*cqproto.GetProviderSchemaResponse
 	ProtocolVersion int
 }
 
-func GetProviderSchema(ctx context.Context, plugin *plugin.Manager, request *GetProviderSchemaOptions) (*ProviderSchema, error) {
-	providerPlugin, err := plugin.CreatePlugin(request.Provider, "", nil)
+func GetProviderSchema(ctx context.Context, manager *plugin.Manager, request *GetProviderSchemaOptions) (*ProviderSchema, error) {
+	providerPlugin, err := manager.CreatePlugin(&plugin.CreationOptions{Provider: request.Provider})
 	if err != nil {
-		log.Error().Str("provider", request.Provider).Err(err).Msg("failed to create provider plugin")
+		log.Error().Stringer("provider", request.Provider).Err(err).Msg("failed to create provider plugin")
 		return nil, err // TODO: should be a diag
 	}
-	defer plugin.ClosePlugin(providerPlugin)
+	defer manager.ClosePlugin(providerPlugin)
```

Comment:
Can we have the plugin implement `io.Closer`? so this would be `defer providerPlugin.Close()`